### ISSUE TITLE
No internal resolution option under new video plugin

### ISSUE THREAD

**Issue by phly95** [p1pkin/demul#362]
Is there a reason the new video plugin does not support internal resolution? Sure this is the most accurate emulator, but internal resolution would be nice.

**Comment by p1pkin** [comment]
it is hard (if possible) to implement within correct renderer and video RAM emulation.
current renderer made it in the way real PoverVR2 works, i.e:
- if 'start render' signal fired - scene is rendered and resulting picture is stored into virtual video memory (which is just 8MB in case of Dreamcast) to area pointed by "Frame Buffer Write Address" hardware register.
- at vertical blank data pointed by "Frame Buffer Read Address" fetched from virtual video memory and shown onscreen.

as you can imagine, if picture will be rendered not native (usually 640x480) but in lets say x8 higher render resolution - it becomes 8*8=64 times larger, its size will be even more than half of whole DC video memory, so if it will be stored to VRAM - it just trashes most of textures and other stuff stored there.
so, there no easy enough way to implement that "internal resolution" thing, as it was in older render plugin versions, where video system was emulated in stripped and hacky way, like in rest of other emulators.

**Comment by phly95** [comment]
OK, that's good you are focusing on accuracy. The most important part is that the game runs without major bugs. I do hope you use your experience with the accurate plugin and find a way around the vram issue (maybe expand vram based on IR, or set separate vram application for rendering?) But you can take your time to make sure it's polished.

**Comment by p1pkin** [comment]
> maybe expand vram based on IR, or set separate vram application for rendering

yep, that's two possible options, both quite hard to implement, and both may appear incompatible (again)  if/then some other currently  not yet unemulated things will be implemented.

**Comment by phly95** [comment]
Yeah, I would say focus on accuracy for now and once you get everything down, then focus on IR.

**Comment by p1pkin** [comment]
honestly ? I'd say no one cares what you would say ;)

**Comment by psennermann** [comment]
If it's not possible to have at least a 2x IR, maybe some sort of anti-aliasing implementation would be helpful, cause nowadays a resolution of 640x480 on 20'' (and more) monitors is really too pixelated.
Other than that maybe it could be renamed "Dx11-native" (and the old one "Dx11old-hires") or something like that, otherwise people will keep on asking/complaining that there is no HD option...

**Comment by Hadocem** [comment]
If it is very difficult to implement internal resolution, then it would be possible, add a Upscale resolution ??

**Comment by RinMaru** [comment]
Not a dev but doesn't it already upscale?

**Comment by ngates87** [comment]
Total idiot here... Would be interesting if something like this would be possible. Don't know the first thing about how to do it. Would the DC bios have to be adjusted to be able to render at a higher res? 

Would be interesting if there could be a way to emulate a faster CPU and more system/vram, (like a Dreacmast Pro edition) but I suppose that would screw some games up. 

Just found this interesting, because recently trying the PS2 emulator (pcsx2), it's really quite impressive how much they are able to clean up the picture quality.  Something similar for DC some day, would be cool. 

- dc fanboy
